VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/InAppRecruitmentScreen/InAppRecruitmentScreen.tsx

```diff
@@ -16,9 +19,30 @@ const { LINK_URL_IN_APP_RECRUITMENT, LINK_URL_VETERAN_USABILITY_PROJECT } = getE
 const InAppRecruitmentScreen: FC<InAppRecruitmentScreenProps> = ({ navigation }) => {
   const { t } = useTranslation(NAMESPACE.COMMON)
   const theme = useTheme()
+  const navigateTo = useRouteNavigation()
+  const currentScreen = useNavigationState((state) => state.routes[state.routes.length - 1])
+  const [isWebviewOpen, setIsWebviewOpen] = useState(false)
+
+  useBeforeNavBackListener(navigation, () => {
+    logAnalyticsEvent(Events.vama_givefb_close())
+  })
+
+  useEffect(() => {
+    // Track when the user is leaving the Webview
+    if (isWebviewOpen && currentScreen.name !== 'Webview') {
+      logAnalyticsEvent(Events.vama_givefb_wv_close())
+      setIsWebviewOpen(false)
+    }
+  }, [isWebviewOpen, currentScreen])
 
   const onPress = () => {
-    navigation.navigate('Webview', { url: LINK_URL_IN_APP_RECRUITMENT, displayTitle: t('webview.vagov'), loadingMessage: t('inAppRecruitment.goToQuestionnaire.loading') })
+    logAnalyticsEvent(Events.vama_givefb_launch())
```

Comment:
I wonder if we should change this analytic to a generic and apply it to more sections than just this with some parameters to differentiate the two